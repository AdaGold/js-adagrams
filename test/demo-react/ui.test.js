import React from 'react';
import App from 'demo-react/ui';
import { render } from 'ink-testing-library';
import { expectRenderToMatchSnapshot } from './expect-render';

describe('Button rendering', () => {
  it('Renders correctly in the default state', () => {
    expectRenderToMatchSnapshot(<App />);
  });

  it.skip('Highlights the next item on right-arrow press', () => {
    const { stdin, lastFrame } = render(<App />);

    stdin.write(String.fromCharCode(13) + String.fromCharCode(10)); // Try to send [Enter]

    expect(lastFrame()).toMatchSnapshot();
  });
});
