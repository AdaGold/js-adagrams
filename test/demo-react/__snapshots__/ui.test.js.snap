// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adagrams App Highlights quit on two right-arrow presses 1`] = `
"┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                  │
│                                                                                                  │
│                         ╭────────────────╮    ╭─────────────╮    [31m╭──────╮[39m                        │
│                         │ Start New Game │    │ How to Play │    [31m│[39m [7m[31mQuit[39m[27m [31m│[39m                        │
│                         ╰────────────────╯    ╰─────────────╯    [31m╰──────╯[39m                        │
│                                                                                                  │
│                                                                                                  │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘"
`;

exports[`Adagrams App Highlights the next item on right-arrow press 1`] = `
"┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                  │
│                                                                                                  │
│                         ╭────────────────╮    ╭─────────────╮    [31m╭──────╮[39m                        │
│                         │ Start New Game │    │ [7mHow to Play[27m │    [31m│[39m [31mQuit[39m [31m│[39m                        │
│                         ╰────────────────╯    ╰─────────────╯    [31m╰──────╯[39m                        │
│                                                                                                  │
│                                                                                                  │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘"
`;

exports[`Adagrams App Quits if quit is selected 1`] = `
"┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                  │
│                                                                                                  │
│                         ╭────────────────╮    ╭─────────────╮    [31m╭──────╮[39m                        │
│                         │ Start New Game │    │ How to Play │    [31m│[39m [7m[31mQuit[39m[27m [31m│[39m                        │
│                         ╰────────────────╯    ╰─────────────╯    [31m╰──────╯[39m                        │
│                                                                                                  │
│                                                                                                  │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
[31mMainMenu: Received 'quit'.[39m"
`;

exports[`Adagrams App Renders correctly in the default state 1`] = `
"┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                  │
│                                                                                                  │
│                         ╭────────────────╮    ╭─────────────╮    [31m╭──────╮[39m                        │
│                         │ [7mStart New Game[27m │    │ How to Play │    [31m│[39m [31mQuit[39m [31m│[39m                        │
│                         ╰────────────────╯    ╰─────────────╯    [31m╰──────╯[39m                        │
│                                                                                                  │
│                                                                                                  │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘"
`;

exports[`Adagrams App Shows the help screen 1`] = `
"╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                                  │
│ Select \\"Start New Game\\" to play.                                                                 │
│                                                                                                  │
│                                                                                                  │
│ Choose the number of players, rounds, and seconds each person has available for guessing.        │
│ Then, enter the name of each player.                                                             │
│                                                                                                  │
│                                                                                                  │
│ Each round, a new set of 10 letters will be chosen. Each player has limited time to form words   │
│ out of the available letters.                                                                    │
│ Whoever has the highest scoring words across all rounds wins the game!                           │
│                                                                                                  │
│                                                                                                  │
│                                            ╭─────────╮                                           │
│                                            │ [7mGo Back[27m │                                           │
│                                            ╰─────────╯                                           │
│                                                                                                  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯"
`;
